// @cdk-uip/react-snackbar/src/Snackbar.js
import React from "react";
import classnames from "classnames";
import { bool, func, node } from "prop-types";

/**
 * Controlled snackbar component.
 */
const Snackbar = ({
  show,
  alignStart,
  message,
  onAction,
  actionText,
  multiline,
  actionOnBottom,
  className,
  ...other
}) => (
  <div
    className={classnames(
      "mdc-snackbar",
      {
        "mdc-snackbar--active": show,
        "mdc-snackbar--align-start": alignStart,
        "mdc-snackbar--multiline": multiline,
        "mdc-snackbar--action-on-bottom": actionOnBottom
      },
      className
    )}
    aria-live="assertive"
    aria-atomic="true"
    aria-hidden="true"
    {...other}
  >
    <div className="mdc-snackbar__text">{message}</div>
    {actionText && (
      <div className="mdc-snackbar__action-wrapper">
        <button
          type="button"
          className="mdc-snackbar__action-button"
          onClick={onAction}
        >
          {actionText}
        </button>
      </div>
    )}
  </div>
);

Snackbar.propTypes = {
  /** Show the snackbar. */
  show: bool.isRequired,
  /** Align to start of screen (tablet and desktop only). */
  alignStart: bool,
  /** The text message to display. */
  message: node,
  /** Called when the action button is clicked. */
  onAction: func,
  /** The text to display for the action button. */
  actionText: node,
  /** Whether to show the snackbar with space for multiple lines of text. */
  multiline: bool,
  /** Whether to show the action below the multiple lines of text. */
  actionOnBottom: bool
};

export default Snackbar;
