// @cdk-uip/react-snackbar/src/TemporarySnackbar.js
import React, { Component } from "react";
import { func, number } from "prop-types";
import Snackbar from "./Snackbar";

/**
 * Used to show a snackbar that is only visible for a specified duratiom.
 */
export default class TemporarySnackbar extends Component {
  static propTypes = {
    /** Timeout duration in ms */
    timeout: number,
    /** Called when the snackbar closes */
    onClose: func,
    /** Called when the snackbar is actioned */
    onAction: func
  };

  static defaultProps = {
    timeout: 2750,
    onClose: () => {},
    onAction: () => {}
  };

  state = { show: false };

  componentDidMount() {
    this.update();
  }

  componentDidUpdate(prevProps) {
    if (prevProps.show !== this.props.show) {
      this.update();
    }
  }

  componentWillUnmount() {
    clearTimeout(this.timer);
  }

  update() {
    if (this.props.show) {
      this.setState({ show: true }, () => {
        this.timer = setTimeout(() => {
          this.setState({ show: false }, () => {
            this.props.onClose();
          });
        }, this.props.timeout);
      });
    }
  }

  onAction = () => {
    clearTimeout(this.timer);
    this.setState({ show: false }, () => this.props.onAction());
  };

  render() {
    const { show, onAction, ...other } = this.props;
    return (
      <Snackbar show={this.state.show} onAction={this.onAction} {...other} />
    );
  }
}
