// @cdk-uip/react-typography/dist/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _Typography = require("./Typography");

Object.defineProperty(exports, "Typography", {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Typography).default;
  }
});

var _Headline = require("./Headline1");

Object.defineProperty(exports, "Headline1", {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Headline).default;
  }
});

var _Headline2 = require("./Headline2");

Object.defineProperty(exports, "Headline2", {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Headline2).default;
  }
});

var _Headline3 = require("./Headline3");

Object.defineProperty(exports, "Headline3", {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Headline3).default;
  }
});

var _Headline4 = require("./Headline4");

Object.defineProperty(exports, "Headline4", {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Headline4).default;
  }
});

var _Headline5 = require("./Headline5");

Object.defineProperty(exports, "Headline5", {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Headline5).default;
  }
});

var _Headline6 = require("./Headline6");

Object.defineProperty(exports, "Headline6", {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Headline6).default;
  }
});

var _Subtitle = require("./Subtitle1");

Object.defineProperty(exports, "Subtitle1", {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Subtitle).default;
  }
});

var _Subtitle2 = require("./Subtitle2");

Object.defineProperty(exports, "Subtitle2", {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Subtitle2).default;
  }
});

var _Body = require("./Body1");

Object.defineProperty(exports, "Body1", {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Body).default;
  }
});

var _Body2 = require("./Body2");

Object.defineProperty(exports, "Body2", {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Body2).default;
  }
});

var _ButtonText = require("./ButtonText");

Object.defineProperty(exports, "ButtonText", {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_ButtonText).default;
  }
});

var _CaptionText = require("./CaptionText");

Object.defineProperty(exports, "CaptionText", {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_CaptionText).default;
  }
});

var _OverlineText = require("./OverlineText");

Object.defineProperty(exports, "OverlineText", {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_OverlineText).default;
  }
});

var _Headline7 = require("./Headline");

Object.defineProperty(exports, "Headline", {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Headline7).default;
  }
});

var _Subheading = require("./Subheading2");

Object.defineProperty(exports, "Subheading2", {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Subheading).default;
  }
});

var _Subheading2 = require("./Subheading1");

Object.defineProperty(exports, "Subheading1", {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Subheading2).default;
  }
});

var _Title = require("./Title");

Object.defineProperty(exports, "Title", {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Title).default;
  }
});

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
