// @cdk-uip/react-typography/dist/Title.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

// Legacy typography support
var Title = function Title(_ref) {
  var adjustMargin = _ref.adjustMargin,
      children = _ref.children,
      other = _objectWithoutProperties(_ref, ["adjustMargin", "children"]);

  return _react2.default.createElement(
    "h2",
    _extends({
      style: {
        fontSize: "1.25rem",
        lineHeight: "2rem",
        fontWeight: 500,
        marginTop: adjustMargin ? 0 : undefined
      }
    }, other),
    children
  );
};

exports.default = Title;
