// @cdk-uip/react-linear-progress/src/LinearProgress.js
import React from "react";
import classnames from "classnames";
import { bool, number } from "prop-types";

const scale = scaleValue => {
  const value = `scaleX(${scaleValue})`;
  return {
    transform: value,
    WebkitTransform: value,
    MozTransform: value,
    OTransform: value,
    MSTransform: value
  };
};

const LinearProgress = ({
  indeterminate,
  reversed,
  className,
  progress,
  closed,
  buffer,
  ...other
}) => (
  <div
    role="progressbar"
    className={classnames(
      "mdc-linear-progress",
      {
        "mdc-linear-progress--indeterminate": indeterminate,
        "mdc-linear-progress--reversed": reversed,
        "mdc-linear-progress--closed": closed
      },
      className
    )}
    {...other}
  >
    <div className="mdc-linear-progress__buffering-dots" />
    <div
      className="mdc-linear-progress__buffer"
      style={scale(indeterminate ? 1 : buffer || 1)}
    />
    <div
      className="mdc-linear-progress__bar mdc-linear-progress__primary-bar"
      style={scale(indeterminate ? 1 : progress)}
    >
      <span className="mdc-linear-progress__bar-inner" />
    </div>
    <div className="mdc-linear-progress__bar mdc-linear-progress__secondary-bar">
      <span className="mdc-linear-progress__bar-inner" />
    </div>
  </div>
);

LinearProgress.propTypes = {
  /** Enable if you do not know how long the process will take */
  indeterminate: bool,
  /** Reverse the direction of the progress bar */
  reversed: bool,
  /** The current progress value (between 0 and 1) */
  progress: number,
  /** Hides the linear progress indicator */
  closed: bool,
  /** Current buffer value (between 0 and 1) */
  buffer: number
};

LinearProgress.defaultProps = {
  progress: 0
};

export default LinearProgress;
