// @cdkglobal/cdk-scripts/config/webpack.config.intl.js
'use strict';

const path = require('path');
const paths = require('./paths');
const config = require('./webpack.config.prod.js');

const buildDir = path.join(paths.appIntlTemp, 'build');
const messagesDir = path.join(paths.appIntlTemp, 'messages');

// Hack the real config
module.exports = Object.assign(config, {
  devtool: false, // don't need source maps
  output: {
    path: buildDir, // do not output a real build, all we need is the messages extracted by the babel plugin
  },
  module: {
    strictExportPresence: true,
    rules: [
      {
        exclude: [/\.(js|jsx)$/],
        loader: require.resolve('file-loader'),
        options: {
          name: 'static/media/[name].[hash:8].[ext]',
        },
      },
      // Just need to process JS with Babel to extract messages
      {
        test: /\.(js|jsx)$/,
        include: paths.appSrc,
        loader: require.resolve('babel-loader'),
        options: {
          babelrc: false,
          presets: [require.resolve('babel-preset-react-app')],
          plugins: [
            [
              require.resolve('babel-plugin-react-intl'),
              { messagesDir: messagesDir },
            ],
          ],
        },
      },
    ],
  },
});
