// @cdk-uip/react-speed-dial/src/SpeedDial.js
import React from "react";
import classnames from "classnames";
import { FAB } from "@cdk-uip/react-fab";
import { string } from "prop-types";

/**
 * Used to create a popup group of mini buttons.
 */
export default class SpeedDial extends React.Component {
  static propTypes = {
    /** Icon to display when the menu is closed */
    icon: string.isRequired
  };

  state = { open: false };

  render() {
    const { className, icon, children, ...other } = this.props;
    return (
      <div
        className={classnames(
          "cdk-speed-dial",
          "cdk-speed-dial--fixed",
          className
        )}
        {...other}
      >
        {this.state.open && (
          <div
            className="cdk-speed-dial__backdrop"
            onClick={() => this.setState({ open: false })}
          />
        )}

        <div className="cdk-speed-dial__container">
          <ul>
            {React.Children.map(children, (child, index) =>
              React.cloneElement(child, {
                exited: !this.state.open,
                onClick: e => {
                  if (this.state.open) {
                    this.setState({ open: false });
                    child.props.onClick && child.props.onClick(e);
                  }
                }
              })
            )}
          </ul>
          <FAB
            onClick={() => {
              this.setState(prevState => ({
                open: !prevState.open
              }));
            }}
          >
            {this.state.open ? "close" : icon}
          </FAB>
        </div>
      </div>
    );
  }
}
