// @cdk-uip/react-app-switcher/src/AppSwitcher.js
/* eslint-disable jsx-a11y/href-no-hash */
import React from "react";
import classnames from "classnames";
import { func } from "prop-types";
import { MDCMenu } from "@material/menu/dist/mdc.menu";

/**
 * Display a grid of available application icons.
 *
 * Use in conjunction with the Global Header components.
 */
export default class AppSwitcher extends React.Component {
  static propTypes = {
    /**
     * Called when the app switcher is cancelled (ESC or clicking outside
     * the application grid).
     */
    onCancel: func
  };

  static defaultProps = {
    onCancel: () => {}
  };

  componentDidMount() {
    if (this.menu) {
      this.MDCComponent = new MDCMenu(this.menu);
      this.MDCComponent.listen("MDCMenu:cancel", this.onCancel);
    }
  }

  componentWillUnmount() {
    if (this.MDCComponent) {
      this.MDCComponent.unlisten("MDCMenu:cancel", this.onCancel);
      this.MDCComponent.destroy();
    }
  }

  componentDidUpdate(prevProps) {
    if (this.MDCComponent) {
      if (this.props.open !== prevProps.open) {
        this.MDCComponent.open = this.props.open;
      }
    }
  }

  onCancel = () => {
    this.props.onCancel();
  };

  render() {
    const { className, children, ...other } = this.props;
    return (
      <div
        className={classnames(
          "mdc-menu",
          "cdk-global-header__app-switcher",
          className
        )}
        tabIndex="-1"
        ref={menu => {
          this.menu = menu;
        }}
        {...other}
      >
        {children}
      </div>
    );
  }
}
