// @cdk-uip/react-app-switcher/dist/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _AppSwitcher = require("./AppSwitcher");

Object.defineProperty(exports, "AppSwitcher", {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_AppSwitcher).default;
  }
});

var _AppSwitcherHome = require("./AppSwitcherHome");

Object.defineProperty(exports, "AppSwitcherHome", {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_AppSwitcherHome).default;
  }
});

var _AppSwitcherItems = require("./AppSwitcherItems");

Object.defineProperty(exports, "AppSwitcherItems", {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_AppSwitcherItems).default;
  }
});

var _AppSwitcherItem = require("./AppSwitcherItem");

Object.defineProperty(exports, "AppSwitcherItem", {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_AppSwitcherItem).default;
  }
});

var _AppSwitcherItemIcon = require("./AppSwitcherItemIcon");

Object.defineProperty(exports, "AppSwitcherItemIcon", {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_AppSwitcherItemIcon).default;
  }
});

var _AppSwitcherItemText = require("./AppSwitcherItemText");

Object.defineProperty(exports, "AppSwitcherItemText", {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_AppSwitcherItemText).default;
  }
});

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
