// @cdk-uip/react-text-area/dist/TextArea.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _classnames = require("classnames");

var _classnames2 = _interopRequireDefault(_classnames);

var _propTypes = require("prop-types");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

/**
 * Multi-line text field that wraps a HTML `<textarea>` element.
 */
var TextArea = function (_React$Component) {
  _inherits(TextArea, _React$Component);

  function TextArea() {
    var _ref;

    var _temp, _this, _ret;

    _classCallCheck(this, TextArea);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = _possibleConstructorReturn(this, (_ref = TextArea.__proto__ || Object.getPrototypeOf(TextArea)).call.apply(_ref, [this].concat(args))), _this), _this.state = { hasFocus: false, isValid: true }, _this.onFocus = function () {
      _this.setState({ hasFocus: true });
    }, _this.onBlur = function (e) {
      _this.setState({ hasFocus: false, isValid: e.target.validity.valid });
    }, _temp), _possibleConstructorReturn(_this, _ret);
  }

  _createClass(TextArea, [{
    key: "render",
    value: function render() {
      var _props = this.props,
          id = _props.id,
          className = _props.className,
          label = _props.label,
          fullWidth = _props.fullWidth,
          dense = _props.dense,
          invalid = _props.invalid,
          value = _props.value,
          disabled = _props.disabled,
          other = _objectWithoutProperties(_props, ["id", "className", "label", "fullWidth", "dense", "invalid", "value", "disabled"]);

      var _state = this.state,
          hasFocus = _state.hasFocus,
          isValid = _state.isValid;

      return _react2.default.createElement(
        "label",
        {
          className: (0, _classnames2.default)("mdc-text-field", "mdc-text-field--textarea", "mdc-text-field--upgraded", {
            "mdc-text-field--focused": hasFocus,
            "mdc-text-field--invalid": invalid || !isValid,
            "mdc-text-field--block": fullWidth,
            "mdc-text-field--dense": dense,
            "mdc-text-field--disabled": disabled
          }, className)
        },
        _react2.default.createElement("textarea", _extends({
          className: "mdc-text-field__input",
          onFocus: this.onFocus,
          onBlur: this.onBlur,
          id: id,
          value: value,
          disabled: disabled
        }, other)),
        label && _react2.default.createElement(
          "span",
          {
            className: (0, _classnames2.default)("mdc-floating-label", {
              "mdc-floating-label--float-above": hasFocus || value
            })
          },
          label
        )
      );
    }
  }]);

  return TextArea;
}(_react2.default.Component);

TextArea.propTypes = {
  /** The text area value */
  value: _propTypes.string,
  /** Called when the value changes */
  onChange: _propTypes.func,
  /** The component ID */
  id: _propTypes.string,
  /** The label text */
  label: _propTypes.node,
  /** Disabled */
  disabled: _propTypes.bool,
  /** Full width */
  fullWidth: _propTypes.bool,
  /** For a more compact version */
  dense: _propTypes.bool,
  /** Display as invalid */
  invalid: _propTypes.bool
};
exports.default = TextArea;
