// @cdk-uip/react-button/src/Button.js
import React from "react";
import classnames from "classnames";
import { bool, func } from "prop-types";

/**
 * - To handle clicks add an `onClick` prop.
 * - Any props not listed will be passed through to the underlying element.
 */
const Button = ({
  raised,
  dense,
  accent,
  className,
  children,
  link,
  primary,
  unelevated,
  outlined,
  buttonRef,
  ...other
}) => {
  const T = link ? "a" : "button";
  return (
    <T
      className={classnames(
        "mdc-button",
        {
          "mdc-button--raised": raised,
          "mdc-button--dense": dense,
          "mdc-button--primary": primary,
          "mdc-button--unelevated": unelevated,
          "mdc-button--outlined": outlined
        },
        className
      )}
      ref={buttonRef}
      {...other}
    >
      {children}
    </T>
  );
};

Button.propTypes = {
  /** Primary button */
  primary: bool,
  /** Raised button. */
  raised: bool,
  /** Dense button. */
  dense: bool,
  /** Render the button as a link */
  link: bool,
  /** Flush with the surface */
  unelevated: bool,
  /** Flush with the surface with a visible border */
  outlined: bool,
  /** Pass a callback ref to get ref of underlying button DOM node */
  buttonRef: func
};

Button.defaultProps = {
  buttonRef: () => {}
};

export default Button;
