// @cdk-uip/react-checkbox/src/Checkbox.js
import React from "react";
import classnames from "classnames";
import { bool } from "prop-types";

/**
 * This is a controlled component. You should supply suitable
 * `checked` and `onChange` props to control the component
 * state.
 *
 * To add a label wrap in a `<FormField>` component.
 */
class Checkbox extends React.Component {
  static propTypes = {
    /** Mark the checkbox as indeterminate */
    indeterminate: bool
  };

  componentDidMount() {
    this.setIndeterminate();
  }

  componentDidUpdate() {
    this.setIndeterminate();
  }

  setIndeterminate() {
    if (this.checkbox) {
      this.checkbox.indeterminate = this.props.indeterminate;
    }
  }

  render() {
    const { className, indeterminate, ...other } = this.props;
    return (
      <div className={classnames("mdc-checkbox", className)}>
        <input
          type="checkbox"
          className="mdc-checkbox__native-control"
          ref={ref => {
            this.checkbox = ref;
          }}
          {...other}
        />
        <div className="mdc-checkbox__background">
          <svg className="mdc-checkbox__checkmark" viewBox="0 0 24 24">
            <path
              className="mdc-checkbox__checkmark-path"
              fill="none"
              stroke="white"
              d="M1.73,12.91 8.1,19.28 22.79,4.59"
            />
          </svg>
          <div className="mdc-checkbox__mixedmark" />
        </div>
      </div>
    );
  }
}

export default Checkbox;
